packages/babel-cli/src/babel/util.ts

```diff
@@ -15,22 +15,54 @@ export function chmod(src: string, dest: string): void {
   }
 }
 
+export function alphasort(a: string, b: string) {
+  return a.localeCompare(b, "en");
+}
+
 type ReaddirFilter = (filename: string) => boolean;
 
 export function readdir(
   dirname: string,
   includeDotfiles: boolean,
   filter?: ReaddirFilter,
 ): Array<string> {
-  return readdirRecursive(dirname, (filename, index, currentDirectory) => {
-    const stat = fs.statSync(path.join(currentDirectory, filename));
-
-    if (stat.isDirectory()) return true;
-
+  if (process.env.BABEL_8_BREAKING) {
     return (
-      (includeDotfiles || filename[0] !== ".") && (!filter || filter(filename))
+      fs
+        .readdirSync(dirname, { recursive: true, withFileTypes: true })
+        .filter(dirent => {
+          // exclude directory entries from readdir results
+          if (dirent.isDirectory()) return false;
+          const filename = dirent.name;
+          return (
+            (includeDotfiles || filename[0] !== ".") &&
+            (!filter || filter(filename))
+          );
+        })
+        .map(dirent => path.join(dirent.path, dirent.name))
+        // readdirSyncRecursive conducts BFS, sort the entries so we can match the DFS behaviour of fs-readdir-recursive
+        // https://github.com/nodejs/node/blob/d6b12f5b77e35c58a611d614cf0aac674ec2c3ed/lib/fs.js#L1421
+        .sort(alphasort)
```

Comment:
Babel cli does not rely on the filenames ordering. The `sort` calls is meant to preserve the test log order. 
If we want to change the ordering in Babel 8, we can remove this and the `glob.sync` usage in `options.ts`.